idl/testing/run_tests.py

```diff
@@ -259,98 +259,127 @@ def build_write_tree(tree, shot):
+        code_lines = list(filter(None, code_lines))
+        code = '\n'.join(code_lines)
+
+        # Running IDL code at the IDL> prompt vs running it as a script causes
+        # weird differences in the evaluation, so we use a test.pro file
+
+        code = 'pro test\n' + code + '\nend'
+        test_file = TEST_DIR.name + '/test.pro'
+        open(test_file, 'wt').write(code)
+
+        expected_lines = [ line.strip() for line in self.expected_output.splitlines() ]
+        expected_lines = list(filter(None, expected_lines))
+
+        log.write(f'Running {self.name}:\n')
+        for line in code_lines:
+            log.write(f'IDL> {line}\n')
+        log.write('\n')
+
+        log.write('Expected:\n')
+        for line in expected_lines:
+            log.write(f'{line}\n')
+        log.write('\n')
+
+        env = dict(os.environ)
+        env['IDL_PATH'] = env['IDL_PATH'] + ':' + TEST_DIR.name
+
+        proc = subprocess.Popen(
+            ['idl'],
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            env=env
+        )
+
+        # We call our test.pro module
+
+        proc.stdin.write('test\nexit\n'.encode())
+        proc.stdin.flush()
+
+        hide_output = True
+
+        log.write('Actual:\n')
+        lines = []
+        while True:
+            line = proc.stdout.readline().decode()
+            if not line:
+                break
+
+            line = line.strip()
+
+            if line is not None and not hide_output:
+                log.write(f'{line}\n')
+                log.flush()
+                if line != '':
+                    lines.append(line.strip())
+
+            # Skip the IDL header and licenseing information, which is everything
+            # above this line
+            if line == '% Compiled module: TEST.':
+                hide_output = False
+        log.write('\n')
+
+        self.passed = True
+
+        for line, expected in zip(lines, expected_lines):
+            if line != expected:
+                log.write(f'`{line}` != `{expected}`\n')
+                self.passed = False
+
+        if self.passed:
+            log.write('Success\n')
+        else:
+            log.write(f'Failure, see {self.logfile}\n')
+
+        log.write('\n')
+        log.write('----------------------------------------------------------------\n')
+        log.write('\n')
+
+        log.close()
+
+        with open(self.logfile, 'rt') as log:
+            print(log.read(), flush=True)
+
+
+all_tests = []
+def idl_test(name, code, expected_output):
+    global all_tests
+
+    print(f'Starting {name}')
+
+    t = IDLTest(name, code, expected_output)
+    t.start()
+    all_tests.append(t)
 
 
 #-------------------------------------------------------------------------------
 # Tree open / read / close
-idl_test(f'''
-         
-testid = 'IDL-tree-read'
+idl_test('IDL-tree-read',
+
```

Comment:
There is a blank line after the `idl_test(IDL-tree-read'`.   Is A-OK "as is".   However, if delete that blank line then this test case has the same style as all the other test cases.